flink/src/test/java/org/apache/iceberg/flink/FlinkTestBase.java

```diff
@@ -78,7 +79,9 @@ protected TableEnvironment getTableEnv() {
               .build();
 
           TableEnvironment env = TableEnvironment.create(settings);
-          env.getConfig().getConfiguration().set(FlinkTableOptions.TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM, false);
+          env.getConfig().getConfiguration()
+              .set(FlinkTableOptions.TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM, false)
+              .set(CoreOptions.DEFAULT_PARALLELISM, 1);
```

Comment:
the default parallelism of the new interface is not 1, It seems to be the number of cores of the current machine，If our current machine is multi-core, some test case will go wrong,for example. 
there is a sql :
`
INSERT INTO mytest VALUES (1,'iceberg',10),(2,'b',20),(3,CAST(NULL AS VARCHAR),30)
`
it will write multiple files,if we use the sql `SELECT * FROM mytest LIMIT 1` to query the table , the result is random ,not the first record , we  will can not do the assertion,  so I set the parallelism is 1.

Reply:
how many tests are affected by the ordering issue? for `LIMIT 1` tests, maybe we shouldn't assert on the actual record that depends on order. just assert on the counts.

Reply:
In my test, all test methods in `org.apache.iceberg.flink.TestFlinkTableSource` that return more than one record of filter push down will be affected. We need to sort the returned results to do assertion.

In addition, in `TestRewriteDataFilesAction#testRewriteLargeTableHasResiduals` , ([link](https://github.com/apache/iceberg/blob/master/flink/src/test/java/org/apache/iceberg/flink/actions/TestRewriteDataFilesAction.java#L251)),I have two `INSERT INTO`, but it generated 8 data files. I haven't look up the reason yet.

I haven't checked whether there are similar problems in other methods. I think  set the parallelism to 1, it is consistent with the original state , it can avoid unknown and unstable results.

Reply:
I am just a little concerned if forcing parallelism of 1 will hide any race condition or not.

Reply:
Let me take a closer look at the default parallelism in the new interface

Reply:
I debug the code , found that the test case will use the parallelism of the `MiniClusterResource` （default 4），In order not to affect more test case, I did not update the parallelism in `FlinkTestBase`.

In [TestFlinkTableSource](https://github.com/apache/iceberg/blob/master/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java) ,many test case will generate multiple files so that the query result is unordered or random, so I override the `getTableEnv` method and set the parallelism to 1 instead of modifying a lot of code to change the logic of the assertion，because in the test class, we just judge whether limit or filter is push down, the number of parallelism has no effect on this